### ISSUE TITLE
Is there an equivalent of `--no-save` for pnpm?

### ISSUE THREAD

**Issue by trusktr** [pnpm/pnpm#1237]
I tried `--no-save` with pnpm, like

```
pnpm install --no-save grpc@1.7.3 --runtime=electron --target=1.7.9
```

but it still added the package to `package.json`. Is it possible to do that without updating package.json?

**Comment by zkochan** [comment]
I guess we could have such option. However, pnpm compares the public `shrinkwrap.yaml` with the private one (in `node_modules/.shrinkwrap.yaml`) and removes all packages that are not present in the public `shrinkwrap.yaml`. So if `grpc` won't be saved into `package.json` then next time you run `pnpm install`, it will be removed from `node_modules`.

@jsumners you were interested in this option as well.

**Comment by jsumners** [comment]
I understand the idea behind removing modules that are not referenced in `package.json`, but there situations where it is very frustrating. Consider an application that is itself a platform that accepts modules as plugins to provide functionality. These plugins are not necessarily required for the application to provide baseline functionality, so they don't belong in `package.json`; they merely need to be added by the user of the application such that the application can find them.

I think a package manager should not assert "the right way" to do things. It should merely be a tool for the developer to manage packages and stay out of the way.

**Comment by zkochan** [comment]
another way would be to install those packages globally. That doesn't work currently though #1210.

**Comment by aleclarson** [comment]
The `--no-save` flag should _not_ update the public/private shrinkwrap. Pretend like PNPM never installed it.

**Comment by bugs181** [comment]
Does peer dependencies option not solve the issue of "plugins"?

https://nodejs.org/en/blog/npm/peer-dependencies/

**Comment by jsumners** [comment]
@bugs181 no. Peer dependencies are a list of known modules that may be installed to enable extra functionality simply by being installed. The situation I described 1) the primary module author has no way of knowing all plugins that will ever exist and 2) each new plugin requires extra configuration and is not simply an "install and done" situation.

**Comment by sramam** [comment]
I have a similar use case to @jsumners. A meta script is generating a workspace. 
The meta script needs a few packages installed locally to manipulate its WIP.
These should not be installed globally, since it pollutes the system.

Currently, the workaround is to install it and uninstall it. Given that I have an
increasing number of workspace module types to be generated, I have to do this 
careful in my meta-script. It would be so much more confidence inspiring if I 
didn't have to worry that this was not missed in the meta script.

**Comment by adbutterfield** [comment]
I have the use case where, we don't want to include bundle analysis packages that we only run locally in the package.json. We would like to be able to install with `--no-save` and run the analysis script.

**Comment by Julian** [comment]
I was trying to use [eslint-nibble](https://github.com/IanVS/eslint-nibble) and ran into this as well.

(Mentioning as "I want to run this transient thing now and never again" seems like a pretty prominent category of things where one wants this kind of behavior).

**Comment by lzxb** [comment]
I also need this feature

**Comment by silverwind** [comment]
This would be useful to install certain dependencies on-demand. When those dependencies host native modules `pnpm install` could fail to compile those native modules, so by only installing them where they are needed, it avoids such failure modes.
